app/src/utils/recordRequest.ts

```diff
@@ -0,0 +1,239 @@
+import type { ChatCompletion, ChatCompletionChunk } from "openai/resources";
+import { type Stream } from "openai/streaming";
+import mergeChunks from "openpipe/openai/mergeChunks";
+import { z } from "zod";
+import { type FineTune, Prisma, UsageType } from "@prisma/client";
+import { captureException } from "@sentry/node";
+import { v4 as uuidv4 } from "uuid";
+
+import { calculateCost } from "~/server/fineTuningProviders/supportedModels";
+import { prisma } from "~/server/db";
+import { typedFineTune } from "~/types/dbColumns.types";
+import { chatCompletionOutput, type chatCompletionInput } from "~/types/shared.types";
+import { posthogServerClient } from "./analytics/serverAnalytics";
+import {
+  countLlamaInputTokens,
+  countLlamaOutputTokens,
+  countOpenAIChatTokens,
+} from "./countTokens";
+import { default as openAIModelProvider } from "~/modelProviders/openai-ChatCompletion";
+import { type BaseModel } from "~/server/fineTuningProviders/types";
+
+export const captureRecordUsage = async ({
```

Comment:
This is basically a wrapper for `recordUsage` that works with streaming, right? Can't we just let `recordUsage` take a completion of type stream or chatcompletion and handle this internally? A separate wrapper I feel like just adds indirection for not much gain in this case.